### PR TITLE
Feature/exit with non zero on command error

### PR DESCRIPTION
**Pull request by Kami**
This branch addresses the following issues: #18, #22

It also includes a bunch of other command line interface related changes and improvements.

Next time I will try to create a separate branch for each logical change :)

### CODE REVIEW

File: lib/cast-client/commands/ca/delete.js [hunk 1]
```diff
@@ -36,27 +38,27 @@ var config = {
...
-  http.getApiResponse(args.remote, '1.0', remotePath, 'DELETE', null, true,
-                      function(err, response) {
-    if (err) {
-      sys.puts('Error: ' + err.message);
-      return;
+  async.series([
+    function(callback) {
+      var remotePath = sprintf('/ca/%s/', args.hostname);
+      http.getApiResponse(args.remote, '1.0', remotePath, 'DELETE', null, true,
+                          [200], function(err, response) {
```
Comment (russellhaering) [inline lib/cast-client/commands/ca/delete.js:45]: This isn't a blocker, but getApiResponse takes way too many arguments.

File: lib/cast-client/commands/bundles/create.js [hunk 2]
```diff
@@ -193,7 +194,7 @@ function handleCommand(args) {
 
   function(err) {
     if (err) {
-      sys.puts('Error: ' + err.message);
+      clientUtils.printErrorAndExit(err, 1);
```
Comment (russellhaering) [inline lib/cast-client/commands/bundles/create.js:197]: I'd rather have handleCommand take a callback that we pass this stuff to. Aside from being what we do most other places, it also always returns control to the same location in a cleaner way IMO. However, with these changes it looks like that should be easy enough to do later, so consider this another non-blocker (get it?)

File: lib/cast-client/commands/ca/delete.js (hunk 1 above)
Comment (Kami) [inline lib/cast-client/commands/ca/delete.js:45]: +1 will open a ticket, it should take an options object and we should have some sane defaults.

File: lib/cast-client/commands/bundles/create.js (hunk 2 above)
Comment (Kami) [inline lib/cast-client/commands/bundles/create.js:197]: Yep, makes sense, so we only need to change it in a single place.

**Pull request merged**